### PR TITLE
[AA-759] fix: don't throw an error when there is no verified mode

### PR DESCRIPTION
**Pull request by MatthewPiatetsky**

### CODE REVIEW

File: src/tab-page/LoadedTabPage.jsx [hunk 1]
```diff
@@ -60,15 +60,17 @@ function LoadedTabPage({
...
-      <StreakModal
-        courseId={courseId}
-        metadataModel={metadataModel}
-        streakLengthToCelebrate={streakLengthToCelebrate}
-        isStreakCelebrationOpen={isStreakCelebrationOpen}
-        closeStreakCelebration={closeStreakCelebration}
-        AA759ExperimentEnabled={AA759ExperimentEnabled}
-        verifiedMode={verifiedMode}
-      />
+      {streakLengthToCelebrate && (
```
Comment (MatthewPiatetsky) [inline src/tab-page/LoadedTabPage.jsx]: add back condition to running the StreakModal component - there isn't a case where we need to run the streak modal code when this value is false (even if the modal is closed)

File: src/shared/streak-celebration/StreakCelebrationModal.jsx [hunk 2]
```diff
@@ -74,18 +74,24 @@ function StreakModal({
...
+  let upgradeUrl;
+  let mode;
+  let offer;
 
-  const offer = {
-    discountedPrice: (mode.price * 0.85).toFixed(2).toString(),
-    originalPrice: mode.price.toString(),
-    upgradeUrl: mode.upgradeUrl,
-  };
+  if (verifiedMode) {
```
Comment (MatthewPiatetsky) [inline src/shared/streak-celebration/StreakCelebrationModal.jsx:84]: all these changes are just coding defensively for when verifiedMode isn't there

**Review by mikix — APPROVED** [review]
Sure. `offer` can be left undefined like that no problem?

File: src/tab-page/LoadedTabPage.jsx [hunk 3]
```diff
@@ -37,12 +37,12 @@ function LoadedTabPage({
 
   const activeTab = tabs.filter(tab => tab.slug === activeTabSlug)[0];
 
+  const { verifiedMode } = useModel(metadataModel, courseId);
```
Comment (ciduarte) [inline src/tab-page/LoadedTabPage.jsx]: can we add this field to the useModel above?

File: src/tab-page/LoadedTabPage.jsx (hunk 1 above)
Comment (ciduarte) [reply src/tab-page/LoadedTabPage.jsx]: nit: I'd rather us early exit within the StreakModal component

**Comment by MatthewPiatetsky** [comment]
offer is set if you've set verified mode
but the stuff using offer only appears if AA759ExperimentEnabled is true and i've added an extra safety check to only set that one to true if verified mode is true

File: src/tab-page/LoadedTabPage.jsx (hunk 1 above)
Comment (MatthewPiatetsky) [reply src/tab-page/LoadedTabPage.jsx]: tbh the existing way seems a tiny bit clearer to me, but sure i've removed this and added an early exit

**Review by ciduarte — APPROVED** [review]

**Pull request merged**